in_toto/runlib.go

```diff
@@ -182,7 +182,11 @@ func recordArtifacts(paths []string, hashAlgorithms []string, gitignorePatterns
 						break
 					}
 				}
-
+				// Check if path is unique
+				_, existingPath := artifacts[path]
+				if existingPath {
+					return fmt.Errorf("Prefix selection has resulted in non unique dictionary key: %s", path)
```

Comment:
```suggestion
					return fmt.Errorf("left stripping has resulted in non unique dictionary key: %s", path)
```

Reply:
I don't know. I think I would prefer to handle the error strings isolated in a new PR. We shouldn't mix code reformatting with bugfixes or features. Commits need to be atomic.

Reply:
I can update the error string and squash both commits if you want.

Reply:
@kadern0 no, not necessary. I really think we should address the wrong error string formatting in a different issue/PR.
Do you agree @adityasaky ?

Reply:
Error string formatting definitely belongs in its own issue.

Reply:
To be clear, @kadern0 I think the suggestion I made (or any other version of including "left strip" in the error message for clarity) is sufficient to merge this PR. Apologies for the non PR specific discussions that may have caused some confusion. I'd only like to update the error message to ensure that anyone using the functionality know _exactly_ what the issue is.

Reply:
Ah okay. Thanks for the clarification. Can we use "Left stripping ..." in this PR then + add a new issue for solving the issues with error formatting in general?

Reply:
I don't think we need to worry about capitalizing "left" only to handle it again in an error strings formatting PR. Since we've (informally) decided to make that change, we can choose to apply it to any changes we make to error strings / new error strings even if the formatting hasn't been applied to the rest. I may have misunderstood you though. :smile:

I also do wonder if we can just handle everything (formatting, using `errors` consistently) under the umbrella of #5.

Reply:
Yep, makes sense to handle this under the umbrella of #5.

> I don't think we need to worry about capitalizing "left" only to handle it again in an error strings formatting PR.

I am a fan of atomic PRs. A PR should consist only of the change necessary for solving the specified problem. Adding reformatting to it adds more "complexitiy". If we merge this PR with one error string starting with lower case, this would be inconsistent with the rest of the code base. It's better to handle this separately, imho. But feel free to merge it.

Reply:
I'm okay either way :smile:

Reply:
In that case, just let us merge this and do the error string reformat under the umbrella of #5 like you suggested.